source/common/http/http1/legacy_parser_impl.cc

```diff
@@ -0,0 +1,209 @@
+#include "common/http/http1/legacy_parser_impl.h"
+
+#include <http_parser.h>
+
+#include "common/common/assert.h"
+#include "common/http/http1/parser.h"
+
+namespace Envoy {
+namespace Http {
+namespace Http1 {
+namespace {
+ParserStatus intToStatus(int rc) {
+  switch (rc) {
+  case -1:
+    return ParserStatus::Error;
+  case 0:
+    return ParserStatus::Success;
+  case 1:
+    return ParserStatus::NoBody;
+  case 2:
+    return ParserStatus::NoBodyData;
+  case 31:
```

Comment:
do we have links of where these magic numbers come from?

Reply:
https://github.com/envoyproxy/envoy/blob/4e71ed698356b68ca4f5ceca6b2fd128e88fe1ac/source/common/http/http1/legacy_parser_impl.cc#L190

Oops, it's linked in `statusToInt`, but not in `intToStatus`. Added